src/Linter.php

```diff
@@ -89,4 +99,43 @@ private function packagesAreSorted(array $packages)
 
         return $sortedNames === $names;
     }
+
+    /**
+     * @param string[] $packages
+     *
+     * @return array
+     */
+    private function validateVersionConstraints(array $packages)
+    {
+        $errors = array();
+
+        foreach ($packages as $name => $constraint) {
+            $errorPrefix = sprintf("Requirement '%s:%s' ", $name, $constraint);
+
+            // Checks if OR format is correct
+            // From Composer\Semver\VersionParser::parseConstraints
+            $orConstraints = preg_split('{\s*\|\|?\s*}', trim($constraint));
+            foreach ($orConstraints as &$subConstraint) {
+                // Checks ~ usage
+                $subConstraint = str_replace('~', '^', $subConstraint);
+
+                // Checks for usage like ^2.1,>=2.1.5. Should be ^2.1.5.
+                // From Composer\Semver\VersionParser::parseConstraints
```

Comment:
I thought "hey let's have a look at Composer's source, see if something can't be reused", then found [this piece of code](https://github.com/composer/semver/blob/master/src/VersionParser.php#L239), then read this comment… I should read comments first, but anyway, good thinking ;)